bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -17,48 +17,593 @@ limitations under the License.
+	//+optional
+	Cni Cni `json:"cni,omitempty"`
+
+	// PauseImage Override image to use for pause.
+	//+optional
+	PauseImage string `json:"pauseImage,omitempty"`
+
+	// RuntimeImage Override image to use for runtime binaries (containerd, kubectl, crictl, etc).
+	//+optional
+	RuntimeImage string `json:"runtimeImage,omitempty"`
+
+	// KubeletPath Override kubelet binary path.
+	//+optional
+	KubeletPath string `json:"kubeletPath,omitempty"`
+
+	//	CloudProviderName  Cloud provider name.
+	//+optional
+	CloudProviderName string `json:"cloudProviderName,omitempty"`
+
+	//	CloudProviderConfigMap  is a reference to a ConfigMap containing Cloud provider configuration.
+	//+optional
+	CloudProviderConfigMap corev1.ObjectReference `json:"cloudProviderConfigMap,omitempty"`
+
+	// NOTE: this was only profile, changed it to cisProfile.
+	// CisProfile Validate system configuration against the selected benchmark (valid items: cis-1.23 ).
+	//+optional
+	CisProfile CisProfile `json:"cisProfile,omitempty"`
+
+	// AuditPolicySecret Path to the file that defines the audit policy configuration.
+	//+optional
+	AuditPolicySecret corev1.ObjectReference `json:"auditPolicySecret,omitempty"`
+
+	// ControlPlaneResourceRequests Control Plane resource requests.
+	//+optional
+	ControlPlaneResourceRequests string `json:"controlPlaneResourceRequests,omitempty"`
+
+	// ControlPlaneResourceLimits Control Plane resource limits.
+	//+optional
+	ControlPlaneResourceLimits string `json:"controlPlaneResourceLimits,omitempty"`
+
+	// Etcd defines optional custom configuration of ETCD.
+	//+optional
+	Etcd ComponentConfig `json:"etcd,omitempty"`
+
+	// KubeApiServer defines optional custom configuration of the Kube API Server.
+	//+optional
+	KubeApiServer ComponentConfig `json:"kubeApiServer,omitempty"`
+
+	// KubeControllerManager defines optional custom configuration of the Kube Controller Manager.
+	//+optional
+	KubeControllerManager ComponentConfig `json:"kubeControllerManager,omitempty"`
+
+	// KubeScheduler defines optional custom configuration of the Kube Scheduler.
+	//+optional
+	KubeScheduler ComponentConfig `json:"kubeScheduler,omitempty"`
+
+	// KubeProxy defines optional custom configuration of the Kube Proxy.
+	//+optional
+	KubeProxy ComponentConfig `json:"kubeProxy"`
+
+	// CloudControllerManager defines optional custom configuration of the Cloud Controller Manager.
+	//+optional
+	CloudControllerManager ComponentConfig `json:"cloudControllerManager,omitempty"`
+
+	// Kubelet defines optional custom configuration of the Kubelet.
+	//+optional
+	Kubelet ComponentConfig `json:"kubelet,omitempty"`
+}
+
+type RKE2AgentConfig struct {
+	// DataDir Folder to hold state.
+	//+optional
+	DataDir string `json:"dataDir,omitempty"`
+
+	// NodeLabels  Registering and starting kubelet with set of labels.
+	//+optional
+	NodeLabels []string `json:"nodeLabels,omitempty"`
```

Comment:
There seems to be some fields in the RKE2AgentConfig struct that are the same from the RKE2ServerConfig struct. We could put these into a common struct and inline it in both RKE2AgentConfig and RKE2ServerConfig